libsql-sqlite3/src/wal.c

```diff
@@ -253,7 +253,22 @@
 #include "wal.h"
 
 typedef libsql_pghdr PgHdr;
-typedef libsql_wal Wal;
+typedef sqlite3_wal Wal;
+
+static int sqlite3WalCheckpoint(
+  Wal *pWal,                      /* Wal connection */
+  sqlite3 *db,                    /* Check this handle's interrupt flag */
+  int eMode,                      /* PASSIVE, FULL, RESTART, or TRUNCATE */
+  int (*xBusy)(void*),            /* Function to call when busy */
+  void *pBusyArg,                 /* Context argument for xBusyHandler */
+  int sync_flags,                 /* Flags to sync db file with (or 0) */
+  int nBuf,                       /* Size of temporary buffer */
+  u8 *zBuf,                       /* Temporary buffer to use */
+  int *pnLog,                     /* OUT: Number of frames in WAL */
+  int *pnCkpt                     /* OUT: Number of backfilled frames in WAL */
+);
+static void sqlite3WalEndReadTransaction(Wal *pWal);
+static int sqlite3WalEndWriteTransaction(Wal *pWal);
```

Comment:
just curious - why were only those 3 forward-declared?

Reply:
because they were the only one that I needed to forward declare for it to compile